gateway/app/schemas/device.py

```diff
@@ -0,0 +1,53 @@
+from typing import Annotated, Optional
+from ipaddress import IPv4Address, IPv6Address
+from pydantic import BaseModel, Field
+
+PhoneNumber = Annotated[
+    str,
+    Field(
+        pattern=r"^\+[1-9][0-9]{4,14}$",
+        description="A public identifier addressing a telephone subscription. In mobile networks it corresponds to the MSISDN (Mobile Station International Subscriber Directory Number). In order to be globally unique it has to be formatted in international format, according to E.164 standard, prefixed with '+'.",
+        examples=["+123456789"],
+    ),
+]
+
+NetworkAccessIdentifier = Annotated[
+    str,
+    Field(
+        description="A public identifier addressing a subscription in a mobile network. In 3GPP terminology, it corresponds to the GPSI formatted with the External Identifier ({Local Identifier}@{Domain Identifier}). Unlike the telephone number, the network access identifier is not subjected to portability ruling in force, and is individually managed by each operator.",
+        examples=["123456789@domain.com"],
+    ),
+]
+
+Port = Annotated[int, Field(description="TCP or UDP port number", ge=0, le=65535)]
+
+SingleIpv4Addr = Annotated[
+    IPv4Address,
+    Field(
+        description="A single IPv4 address with no subnet mask",
+        examples=["84.125.93.10"],
+    ),
+]
+
+
+class DeviceIpv4Addr(BaseModel):
+    publicAddress: SingleIpv4Addr
+    privateAddress: SingleIpv4Addr
+    publicPort: Optional[Port] = None
```

Comment:
I don't know how you generated this scheme, but it isn't correct. The only required field is `publicAddress`, the other two are optional, but at least one must be set.